src/poetry/core/packages/dependency_group.py

```diff
@@ -3,30 +3,38 @@
 from collections import defaultdict
 from typing import TYPE_CHECKING
 
+from packaging.utils import NormalizedName
+from packaging.utils import canonicalize_name
+
 from poetry.core.version.markers import parse_marker
 
 
 if TYPE_CHECKING:
     from poetry.core.packages.dependency import Dependency
     from poetry.core.version.markers import BaseMarker
 
-MAIN_GROUP = "main"
+MAIN_GROUP: NormalizedName = canonicalize_name("main")
 
 
 class DependencyGroup:
     def __init__(
         self, name: str, *, optional: bool = False, mixed_dynamic: bool = False
     ) -> None:
-        self._name: str = name
+        self._name: NormalizedName = canonicalize_name(name)
+        self._pretty_name: str = name
         self._optional: bool = optional
         self._mixed_dynamic = mixed_dynamic
         self._dependencies: list[Dependency] = []
         self._poetry_dependencies: list[Dependency] = []
 
     @property
-    def name(self) -> str:
+    def name(self) -> NormalizedName:
         return self._name
 
+    @property
+    def pretty_name(self) -> str:
```

Comment:
Is `pretty_name` the clearest name for this property? I'd expect something like `original_name`.

I can undestand this causing confusion for folks (what property should I use?), but I think you've mitigated that by preferring `NormalizedName` in API signatures.

Reply:
(I see that's aligned with `Package`)

Reply:
Yes, I chose the name for consistency, which is more important to me than the actual name. If we decided that `pretty_name` is not a good name, we should deprecate it in `Package` and introduce the new name there, too. (In my opinion, `pretty_name` is not that bad that it is worth the trouble.)

Reply:
Seems good to me :)